### DISCUSSION TITLE
Switching version control wipes translations

### DISCUSSION THREAD

**Discussion by maxxer** [WeblateOrg/weblate#9348] [Q&A]
Hi.
We have some PHP projects currently using PO files. We now would like to switch to PHP file format for translation files, but when we do so, we create new *empty* files (and delete old `.po`s) and commit, thus Weblate picks those empty strings and wipes the existing translation.

We hoped strings could be stored into Weblate's db, thus being populated automatically from previous PO files, but this isn't the case.
We tried exporting translation memory before the switch, and reimporting it right after push, but apparently didn't work.

Are we doing something wrong, or is it not currently possible to retain translations when switching version control?

thanks

**Comment by nijel** [comment]
Weblate is in the end interface for editing files. So if you start with empty files, the translations will be empty. If you have the translation memory, you can use automatic translation to fill in the translations from it.

**Reply by maxxer** [reply]
Thanks. As said, I tried saving and reimporting translation memory, but apparently it didn't do anything. Unless, I didn't wait enough for it to populate? How long can it take?

**Reply by nijel** [reply]
If you import the translation memory, you still need to apply it, for example using https://docs.weblate.org/en/latest/user/translating.html#auto-translation.
